src/main/java/com/cburch/logisim/fpga/hdlgenerator/HDL.java

```diff
@@ -69,6 +127,63 @@ public static String xorOperator() {
     return isVHDL() ? " XOR " : "^";
   }
 
+  public static String addOperator(String signalOne, String signalTwo, boolean signed) {
+    return (isVHDL() ? "STD_LOGIC_VECTOR(" : "")
+            + typecast(signalOne, signed)
+            + " + "
+            + typecast(signalTwo, signed)
+            + (isVHDL() ? ")" : "");
+  }
+  
+  public static String subOperator(String signalOne, String signalTwo, boolean signed) {
+    return (isVHDL() ? "STD_LOGIC_VECTOR(" : "")
+            + typecast(signalOne, signed)
+            + " - "
+            + typecast(signalTwo, signed)
+            + (isVHDL() ? ")" : "");
+  }
+
+  public static String shiftlOperator(String signal, String nrOfBits, boolean arithmetic) {
+    return isVHDL() ? "STD_LOGIC_VECTOR(SHIFT_LEFT(" + typecast(signal, arithmetic) + ", " + nrOfBits + "))"
+                    : "(" + signal + (arithmetic ? " <<< " : " << ") + nrOfBits + ")";
+  }
+  
+  public static String shiftrOperator(String signal, String nrOfBits, boolean arithmetic) {
+    return isVHDL() ? "STD_LOGIC_VECTOR(SHIFT_RIGHT(" + typecast(signal, arithmetic) + ", " + nrOfBits + "))"
+                    : "(" + signal + (arithmetic ? " >>> " : " >> ") + nrOfBits + ")";
```

Comment:
I would really prefer here vector concatenation operations, as the number of shifts is fixed.

Reply:
I'm working on this.
Shift left and rotate are no problem.
But I still need to figure out the best way for arithmetic shift right.

Reply:
In VHDL:

```
shifted_vector(31 DOWNTO 24) <= (OTHERS => orig_vector(31));
shifted_vector(23 DOWNTO  0) <= orig_vector(31 DOWNTO 8);
```

in Verilog:
```
shifted_vector = {8{orig_vector[31]}, orig_vector[31:8]}}
```

@MarcinOrlowski :
Now I found where the ```{``` and ```}```  are used, in Verilog for vector concatination, hence we have to stay with the double ones :-)

Reply:
Yes, that's why I asked first :)